security/pkg/pki/util/generate_csr_test.go

```diff
@@ -25,80 +26,126 @@ import (
 
-	if err != nil {
-		t.Errorf("failed to gen CSR")
-	}
+		if err != nil {
+			t.Errorf("%s: failed to gen CSR", id)
+		}
 
-	pemBlock, _ := pem.Decode(csrPem)
-	if pemBlock == nil {
-		t.Fatalf("failed to decode csr")
-	}
-	csr, err := x509.ParseCertificateRequest(pemBlock.Bytes)
-	if err != nil {
-		t.Fatalf("failed to parse csr")
-	}
-	if err = csr.CheckSignature(); err != nil {
-		t.Errorf("csr signature is invalid")
-	}
-	if csr.Subject.Organization[0] != "MyOrg" {
-		t.Errorf("csr subject does not match")
-	}
-	if !strings.HasSuffix(string(csr.Extensions[0].Value), "test_ca.com") {
-		t.Errorf("csr host does not match")
+		pemBlock, _ := pem.Decode(csrPem)
+		if pemBlock == nil {
+			t.Fatalf("%s: failed to decode csr", id)
+		}
+		csr, err := x509.ParseCertificateRequest(pemBlock.Bytes)
+		if err != nil {
+			t.Fatalf("%s: failed to parse csr", id)
+		}
+		if err = csr.CheckSignature(); err != nil {
+			t.Errorf("%s: csr signature is invalid", id)
+		}
+		if csr.Subject.Organization[0] != "MyOrg" {
+			t.Errorf("%s: csr subject does not match", id)
+		}
+		if !strings.HasSuffix(string(csr.Extensions[0].Value), "test_ca.com") {
+			t.Errorf("%s: csr host does not match", id)
+		}
+		if tc.csrOptions.IsEC {
+			if reflect.TypeOf(csr.PublicKey) != reflect.TypeOf(&ecdsa.PublicKey{}) {
+				t.Errorf("%s: decoded PKCS#8 returned unexpected key type: %T", id, csr.PublicKey)
+			}
+		} else {
+			if reflect.TypeOf(csr.PublicKey) != reflect.TypeOf(&rsa.PublicKey{}) {
+				t.Errorf("%s: decoded PKCS#8 returned unexpected key type: %T", id, csr.PublicKey)
+			}
+		}
 	}
 }
 
 func TestGenCSRPKCS8Key(t *testing.T) {
 	// Options to generate a CSR.
-	csrOptions := CertOptions{
-		Host:       "test_ca.com",
-		Org:        "MyOrg",
-		RSAKeySize: 512,
-		PKCS8Key:   true,
+	cases := map[string]struct {
+		csrOptions CertOptions
+	}{
+		"PKCS8Key with RSA": {
+			csrOptions: CertOptions{
+				Host:       "test_ca.com",
+				Org:        "MyOrg",
+				RSAKeySize: 2048,
+				PKCS8Key:   true,
+			},
+		},
+		"PKCS8Key with EC": {
+			csrOptions: CertOptions{
+				Host:     "test_ca.com",
+				Org:      "MyOrg",
+				IsEC:     true,
+				PKCS8Key: true,
+			},
+		},
 	}
 
-	csrPem, keyPem, err := GenCSR(csrOptions)
+	for id, tc := range cases {
+		csrPem, keyPem, err := GenCSR(tc.csrOptions)
 
-	if err != nil {
-		t.Errorf("failed to gen CSR")
-	}
+		if err != nil {
+			t.Errorf("%s: failed to gen CSR", id)
+		}
 
-	pemBlock, _ := pem.Decode(csrPem)
-	if pemBlock == nil {
-		t.Fatalf("failed to decode csr")
-	}
-	csr, err := x509.ParseCertificateRequest(pemBlock.Bytes)
-	if err != nil {
-		t.Fatalf("failed to parse csr")
-	}
-	if err = csr.CheckSignature(); err != nil {
-		t.Errorf("csr signature is invalid")
-	}
-	if csr.Subject.Organization[0] != "MyOrg" {
-		t.Errorf("csr subject does not match")
-	}
-	if !strings.HasSuffix(string(csr.Extensions[0].Value), "test_ca.com") {
-		t.Errorf("csr host does not match")
-	}
+		pemBlock, _ := pem.Decode(csrPem)
```

Comment:
Nit: spacing if off

Reply:
Part of a table test now. The linter would catch this anyway.